common/lib/core-interfaces/src/provider.ts

```diff
@@ -0,0 +1,72 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * @internal
+ * This utility type is primarily meant for internal use by @see Provider
+ * Produces a valid Provider key given a type and a property.
+ * A valid Provider key is a property that exists on the incoming type
+ * as well as on the type of the property itself. For example, IProvideFoo.IFoo.IFoo
+ * This aligns with the provider pattern expected to be used with all Providers.
+ * For example:
+ * ```
+ * interface IProvideFoo{
+ *  IFoo: IFoo
+ * }
+ * interface IFoo extends IProvideFoo{
+ *  foobar();
+ * }
+ * ```
+ * This pattern enables discovery, and delegation in a standard way which is central
+ * to Provider pattern
+ */
+ export type ProviderPropertyKeys<T, TProp extends keyof T = keyof T> =
+    string extends TProp ? never : number extends TProp? never : // exclude indexers [key:string |number]: any
+    TProp extends keyof T[TProp] // TProp is a property of T, T[TProp] and, T[TProp][TProp]
+        ? TProp extends keyof T[TProp][TProp] // ex; IProvideFoo.IFoo.IFoo.IFoo
```

Comment:
Why is this case necessary? It works fine for me if I ditch it. Seems like the magic here is to say "show me my props where the prop name also exists on the prop's type" (line 27), and that is true for both IFoo and IProvideFoo.

Reply:
@anthony-murphy do your tests pass if you remove this case?

Reply:
not seeing problems. so removing this, as it doesn't seem necessary